src/Foundation/NSDate.cs

```diff
@@ -42,10 +42,19 @@ public partial class NSDate {
 		public static explicit operator DateTime (NSDate d)
 		{
 			double secs = d.SecondsSinceReferenceDate;
-			if ((secs < -63113904000) || (secs > 252423993599))
-				throw new ArgumentOutOfRangeException ("Value is outside the range of NSDate");
 
-			return new DateTime ((long)(secs * TimeSpan.TicksPerSecond + NSDATE_TICKS), DateTimeKind.Utc);
+			if ((secs < -63114076800) || (secs > 252423993599))
+				throw new ArgumentOutOfRangeException ($"Value is outside the range of NSDate {secs}");
+
+			var calendar = new NSCalendar (NSCalendarType.Gregorian);
+			calendar.TimeZone = NSTimeZone.FromName ("UTC");
+			NSDateComponents components = calendar.Components (NSCalendarUnit.Year | NSCalendarUnit.Month | NSCalendarUnit.Day | NSCalendarUnit.Hour |
+				NSCalendarUnit.Minute | NSCalendarUnit.Second | NSCalendarUnit.Nanosecond | NSCalendarUnit.Calendar, d);
+
+			var retDate = new DateTime ((int) components.Year, (int) components.Month, (int) components.Day, (int) components.Hour,
+				(int) components.Minute, (int) (components.Second), Convert.ToInt32 (components.Nanosecond / 1000000.0), DateTimeKind.Utc);
```

Comment:
is 1000000.0 a constant? Number of ticks maybe?

Reply:
Unfortunately I wasn't able to find a constant that matched. I've added a constant to the `NSDate` class instead :)